chainerx_cc/chainerx/routines/binary.cc

```diff
@@ -122,4 +142,28 @@ Array BitwiseXor(const Array& x1, Scalar x2) {
 
 Array BitwiseXor(Scalar x1, const Array& x2) { return BitwiseXor(x2, x1); }
 
+Array LeftShift(const Array& x1, const Array& x2) {
+    CheckBitwiseDtypes(x1, x2);
+    return internal::BroadcastBinary(BitwiseImpl<LeftShiftKernel>, x1, x2, ResultType(x1, x2));
+}
+
+Array LeftShift(const Array& x1, Scalar x2) {
+    CheckBitwiseDtypes(x1, x2);
+    return internal::Binary(BitwiseASImpl<LeftShiftASKernel>, x1, x2, ResultType(x1, x2));
+}
+
+Array LeftShift(Scalar x1, const Array& x2) { return LeftShift(x2, x1); }
```

Comment:
I guess this is wrong as `<<` is not commutative i.e. `2 << 5 != 5 << 2`.

It is ok for `&` or `|` because, `2&5 = 5&2`.

Reply:
Now I fix this bug. Is this solution ok for you?

Reply:
Yes looks good to me on a quick glance.